File: testdata/csharp/intelligence/negative.cs
using Microsoft.AspNetCore.Http;
using Microsoft.EntityFrameworkCore;
using System.Diagnostics;
using System.IO;
using System.Net.Http;
using System.Threading.Tasks;
class Handler {
  async Task Run(HttpRequest request, DbContext db) {
    var value = request.Query["value"].ToString();
    db.Database.ExecuteSqlRaw("DELETE FROM Users WHERE Id = {0}", value);
    Process.Start("tool.exe", value);
    File.ReadAllText("/var/app/static.txt");
    await new HttpClient().GetAsync("https://example.com");
    await Task.Delay(1);
    var handler = new HttpClientHandler();
    handler.ServerCertificateCustomValidationCallback = ValidateCertificate;
  }
}

File: testdata/rust/intelligence/negative.rs
use axum::extract::Query;
use sqlx;
use reqwest;
use tokio;
async fn handler(Query(params): Query<Params>) {
    tokio::task::spawn_blocking(|| std::thread::sleep(std::time::Duration::from_secs(1))).await;
    sqlx::query("SELECT * FROM users WHERE id = $1").bind(params.id).await;
    std::process::Command::new("tool").arg(&params.value).status();
    std::fs::read_to_string("/var/app/static.txt");
    reqwest::get("https://example.com").await;
    reqwest::Client::builder().danger_accept_invalid_certs(false);
}
